### DISCUSSION TITLE
用户重复评论问题

### DISCUSSION THREAD

**Discussion by xiamuguizhi** [ArtalkJS/Artalk#134] [Q&A] [answered]
我知道一般遇到这个概率的比较少，但是偶尔还是会发生。

就比如昨天我回复评论，发送评论愣是点了两次，结果就是评论重复了和邮件通知发送了两封。

能否在发送评论后限制一下，避免短时间重复 点击 触发。。

**Comment by xiamuguizhi** [comment]
点击某个按钮时如果点击的比较快，可能会触发多次。

**Reply by qwqcode** [reply]
前端也有一个 BUG，已修复

由于 Loading 显示无效，而导致的评论重复发送问题 https://github.com/ArtalkJS/Artalk/commit/2770c13a56f462b31a2b3dac7f213db512ead2c2

**Comment by qwqcode** [comment]
点击发送评论，会弹出加载动画，会遮盖掉发送按钮，所以用户无法点击。所以应该不存在这种情况吧

![image](https://user-images.githubusercontent.com/22412567/166185332-c1e13ee7-3866-4da2-baaa-a05835ef455e.png)

**Reply by xiamuguizhi** [reply]
诶，难道是我的修改版css 屏蔽了？我没看到这个动画！！！

那应该是我的问题了。

**Reply by xiamuguizhi** [reply]
@qwqcode 

又遇到了，改css 应该影响不到吧。。。。

![image](https://user-images.githubusercontent.com/19650297/166236892-58e31a4c-860b-4b5e-9f39-35e044823879.png)

**Reply by qwqcode** [reply]
奇怪🤔

**Reply by qwqcode** [reply]
确定这个评论是反复点击“发送评论”按钮造成的吗？

**Reply by xiamuguizhi** [reply]
@qwqcode 
又试了一下，触发那个转圈圈了。

真的是奇怪了，难道是提交得到时候卡住了，以为没反应然后在点击一个发送按钮，出发了评论两次！！

估计和网络延迟有关系吧！！！可能还是我自己这边的问题。

![image](https://user-images.githubusercontent.com/19650297/166345945-1994aded-b720-4b28-8a6f-f42c6782c5bd.png)

**Reply by qwqcode** [reply]
我推测可能是验证码有点bug

**Reply by qwqcode** [reply]
2.2.8之后的版本修改了评论发送的代码，还没来得及测试

**Reply by qwqcode** [reply]
测试了不是验证码的问题，我这边没有复现过这个bug，还是不知道什么原因

**Reply by qwqcode** [reply]
这个问题确实在某些情况下存在，v2.3.2 已修复

**Comment by qwqcode** [comment]
刚刚测试复现了这个 BUG：发送一条新评论，然后回复自己的评论，然后再回复自己的评论，刷新页面

正在修

**Reply by qwqcode** [reply]
刚刚复现了两次，但后来就又没问题了 奇怪

**Reply by qwqcode** [reply]
查明了，和删除评论操作有关 https://github.com/ArtalkJS/Artalk/issues/165

**Comment by qwqcode** [comment, marked as answer]
已修复 https://github.com/ArtalkJS/ArtalkGo/commit/c478608b1b00999965cd161b85f3628569a22cd1

**Comment by syhily** [comment]
个人认为，重复评论是否应该在发表评论时进行检测，然后接口直接报错？
